firebase-query.html

```diff
@@ -322,7 +343,14 @@
             this.async(function() {
               var index = this.__indexFromKey(key);
               var value = snapshot.val();
-              value.$key = key;
+              var leaf = typeof value !== 'object';
+
+              if (leaf) {
+                value = {$key: key, $val: value};
+              } else {
+                value.$key = key;
+              }
```

Comment:
Might be nice to factor this into a conversion function since we do the check three times.